org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/format/LSPFormatFilesHandler.java

```diff
@@ -81,15 +82,21 @@ protected void formatFile(final @NonNull IFile file, final IProgressMonitor moni
 				return;
 
 			monitor.setTaskName(NLS.bind(Messages.LSPFormatFilesHandler_FormattingFile, file.getFullPath()));
-			final VersionedFormatRequest formatRequest = formatter.versionedRequestFormatting(doc,
-					new TextSelection(0, 0));
+			final var formatting = formatter.requestFormatting(doc,
+					new TextSelection(0, 0), batchToken).get();
```

Comment:
should we not have a timeout on this get() operation?

Reply:
One for @sebthom : his original submission just used `.get()` so I didn't change that. https://github.com/eclipse/lsp4e/blob/53defa0606b10b5a87d74e5cc5e41a6c1ae0e800/org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/format/LSPFormatFilesHandler.java#L87

Reply:
Happy to put one in if you both agree.

Reply:
We need to put one if we do not want to risk the thread hanging forever, don't we? Recently all the get() where replaced in the code base with get(timeout), so I think that is the standard unless there are very good reasons